tests/metrics/regression/test_psnr.py

```diff
@@ -14,32 +14,55 @@
 
 Input = namedtuple('Input', ["preds", "target"])
 
+_input_size = (NUM_BATCHES, BATCH_SIZE, 32, 32)
 _inputs = [
     Input(
-        preds=torch.randint(n_cls_pred, (NUM_BATCHES, BATCH_SIZE), dtype=torch.float),
-        target=torch.randint(n_cls_target, (NUM_BATCHES, BATCH_SIZE), dtype=torch.float),
+        preds=torch.randint(n_cls_pred, _input_size, dtype=torch.float),
+        target=torch.randint(n_cls_target, _input_size, dtype=torch.float),
     ) for n_cls_pred, n_cls_target in [(10, 10), (5, 10), (10, 5)]
 ]
 
 
-def _sk_metric(preds, target, data_range):
-    sk_preds = preds.view(-1).numpy()
-    sk_target = target.view(-1).numpy()
-    return peak_signal_noise_ratio(sk_target, sk_preds, data_range=data_range)
+def _to_sk_peak_signal_noise_ratio_inputs(value, dim):
+    value = value.numpy()
+    batches = value[None] if value.ndim == len(_input_size) - 1 else value
 
+    if dim is None:
+        return [batches]
 
-def _base_e_sk_metric(preds, target, data_range):
-    sk_preds = preds.view(-1).numpy()
-    sk_target = target.view(-1).numpy()
-    return peak_signal_noise_ratio(sk_target, sk_preds, data_range=data_range) * np.log(10)
+    num_dims = np.size(dim)
+    if not num_dims:
+        return batches
+
+    inputs = []
+    for batch in batches:
+        batch = np.moveaxis(batch, dim, np.arange(-num_dims, 0))
+        psnr_input_shape = batch.shape[-num_dims:]
+        inputs.extend(batch.reshape(-1, *psnr_input_shape))
+    return inputs
+
+
+def _sk_metric(preds, target, data_range, dim):
+    sk_preds_lists = _to_sk_peak_signal_noise_ratio_inputs(preds, dim=dim)
+    sk_target_lists = _to_sk_peak_signal_noise_ratio_inputs(target, dim=dim)
+    return np.mean([
```

Comment:
if the `reduction` argument suddenly have an effect now, it should also be tested. Hopefully it comes down to changing the `np.mean`:
```suggestion
    reduction = np.mean if reduction=='mean' else np.sum
    return np.mean([
```

Reply:
I'm not sure how to deal with the test case `ddp=True, reduction='none'` properly. The order of synchronized _PSNR_ scores from all workers seems to be random.

Reply:
i am fine with just testing `elementwise_mean` and `sum` :]